### ISSUE TITLE
clarify requirement of -keyfile in signing of CSR

### ISSUE THREAD

**Issue by dbist** [cockroachdb/docs#7228]
This is in place of tracking feedback from https://github.com/cockroachdb/docs/pull/7104. Customer reports they do not have access to CA key file when they generate node and user certs. The command `openssl ca -config ca.cnf -keyfile my-safe-directory/ca.key -cert certs/ca.crt -policy signing_policy -extensions signing_node_req -out certs/node.crt -outdir certs/ -in node.csr -batch` is not possible for them. Can the CSR be signed and CRT generated w/out passing `-keyfile my-safe-directory/ca.key`?

**Comment by dbist** [comment]
@Amruta-Ranade  @aaron-crl  @petermattis @drewdeally  FYI

**Comment by aaron-crl** [comment]
@dbist I may not be understanding your question but it sounds like you are asking if the customer can generate node and client certificate signing requests and then send them to an external CA for signing instead of using the local one from the docs process?

Is this correct?

I also don't see how this relates to #7104

**Comment by dbist** [comment]
From the slack discussion:

```
Ben: the thing that signs a cert must by definition have the CA key. if the key lives in a hardware security module, then they'll need to use tools that can talk to their HSM instead of the ones we provide

Artem: so then the equivalent command to openssl ca -config ca.cnf -keyfile my-safe-directory/ca.key -cert certs/ca.crt -policy signing_policy -extensions signing_node_req -out certs/node.crt -outdir certs/ -in node.csr -batch will replace -keyfile with an API call to get the key?
New

Raphael: usually hardware devices come with replacements to the openssl tool to perform certificate operations including signing
they should refer to their tool's documentation
```

**Comment by dbist** [comment]
@aaron-crl my question was based on assumption whether the command above would work without specifying key altogether. I was able to get the answer that the CA auth will provide a key in some form whether from a filesystem location or a hardware module. I included the other issue to ideate whether there's an opportunity to enhance the docs with examples of pulling the CA key from a location other than file system?
